### PR TITLE
Change nan literal syntax to avoid including parens.

### PR DESCRIPTION
**Pull request by sunfishcode**
Change from `nan(0x...)` to `nan:0x...`, since parens are reserved
characters in the s-expression syntax.

### CODE REVIEW

**Comment by kripken** [comment]
This looks like a nice improvement.

**Comment by drom** [comment]
:+1:

File: ml-proto/spec/float.ml [hunk 1]
```diff
@@ -143,9 +143,9 @@ struct
 
   let of_signless_string x len =
     if x <> "nan" &&
-         (len > 7) &&
-           (String.sub x 0 6) = "nan(0x" && (String.get x (len - 1)) = ')' then
-      (let s = Rep.of_string (String.sub x 4 ((len - 5))) in
+         (len > 6) &&
+           (String.sub x 0 6) = "nan:0x" then
+      (let s = Rep.of_string (String.sub x 4 ((len - 4))) in
```
Comment (rossberg-chromium) [inline ml-proto/spec/float.ml]: Nit: there's an awful lot of parens around the `len - 4`. :)

**Comment by rossberg-chromium** [comment]
LGTM

**Comment by sunfishcode** [comment]
Redundant parens removed. Thanks!

**Comment by drom** [comment]
Thank you. I have added new `nan` format support into https://github.com/drom/wast-parser

**Pull request merged**